spec/javascript/support/puppeteer/steps/supplementalApplicationSteps.js

```diff
@@ -29,6 +29,44 @@ const testStatusModalUpdate = async (page) => {
   expect(latestComment).toBe(COMMENT)
 }
 
+const generateRandomCurrency = () => {
+  const val = (Math.random() * 1000).toFixed(2)
+  // TODO We could improve this by adding commas to the currency string too.
+  return {'currency': `$${val}`, 'float': val}
+}
+
+const enterValue = async (page, selector, value) => {
+  // Wait for the field to appear
+  await page.waitForSelector(selector)
+  // Clear the value that's there
+  await page.$eval(selector, (el) => { el.value = '' })
+  // Enter the value
+  await page.type(selector, value)
+}
+
+const getValue = async (page, selector) => {
+  await page.waitForSelector(selector)
+  const input = await page.$(selector)
+  const valueHandle = await input.getProperty('value')
+  return valueHandle.jsonValue()
+}
+
+const savePage = async (page) => {
+  const selector = '#save-supplemental-application'
+  // To allow waitForNavigation to work post-save, we need to scroll
+  //   down to the save button, then wait for the url to update to the scrollable anchor
+  // FIXME this will cause issues if we use savePage when we're already at the
```

Comment:
The issue here:
If the page scrolls down to the save button, the url updates to add #status-history-section. Without this step the timing was such that my page.waitForNavigation() after save was just catching this url update rather than the actual page reload. 

I tried importing goToAnchor from 'react-scrollable-anchor' and it didn't appear to do anything.
I also tried doing page.goto('#status-history-section') and that didn't work either. 

Let me know if you have any other ideas of how to avoid the waitForNavigation here.